lib/dataset/test/bin_test.cpp

```diff
@@ -707,3 +707,53 @@ TEST(BinLinspaceTest, many_events_many_bins) {
       linspace(0.0 * units::one, 1.0 * units::one, Dim::X, 70000);
   EXPECT_EQ(sum(bin(da, {edges}).data()), sum(da.data()));
 }
+
+TEST(BinTest, noncontiguous_edges) {
+  const auto table = make_table(10);
+  const auto cont_x =
+      makeVariable<double>(Dims{Dim::X}, Shape{3}, Values{-2, 0, 2});
+  EXPECT_TRUE(cont_x.stride(Dim::X) == 1);
+
+  const auto x_edges =
+      makeVariable<double>(Dims{Dim::X}, Shape{5}, Values{-2, -1, 0, 1, 2});
+  const scipp::core::Slice slice(Dim::X, 0, 5, 2);
+  const auto non_cont_x = x_edges.slice(slice);
+  EXPECT_TRUE(non_cont_x.stride(Dim::X) != 1);
+
+  EXPECT_EQ(bin(table, {non_cont_x}), bin(table, {cont_x}));
+}
```

Comment:
Can you check in all cases that the coord of the input still references the "non-cont" variable provided by the caller, and NOT the copy made internally.

Reply:
I still couldn't figure out how to access to the original reference...

Reply:
Something like
```python
EXPECT_EQ(&result.coords()[Dim::X].values<double>()[0], x_edges.values<double>()[0])
```
might work.